### DISCUSSION TITLE
Section 4 Detecting Smart Contract vs EOA

### DISCUSSION THREAD

**Discussion by shobhit-gupta** [Cyfrin/security-and-auditing-full-course-s23#80] [Q&A]
Section 4 - Lesson 60: Reporting: Smart Contract Wallet Reverts Winning (If the winner doesn't implement `receive` or `fallback` function)

> 1. Do not allow smart contract entrants (not recommended)

How can a smart contract (such as PuppyRaffle) enforce this? Is there a way in Solidity to determine if a given address is a Smart Contract or an EOA (Externally Owned Account)?

**Comment by lrivo** [comment]
In the `Address` library from OpenZeppelin you can use the method `isContract(address account)` that returns `true` if the passed address is a contract.
The function was removed in the latest version of the OpenZeppelin contracts (the 5.0) but PuppyRaffle uses version 3.4 so you can still use this function.
https://docs.openzeppelin.com/contracts/3.x/api/utils#Address

**Reply by TilakMaddy** [reply]
https://solidity-by-example.org/hacks/contract-size/
